packages/shared/src/utils/array/array.ts

```diff
@@ -1,6 +1,16 @@
 export const shuffleArray = <T>(array: T[]): T[] => {
+    function firstDigit(num: number) {
+        const matches = String(num).match(/\d/) || '1';
+        const digit = Number(matches[0]);
+        return num < 0 ? -digit : digit;
```

Comment:
since we have `Uint32Array` at line 10, there is no way the num is going to be less than 0. The random number is always going to be positive. `num < 0` is an unnecessary check here.